ydb/core/tx/replication/service/transfer_writer.cpp

```diff
@@ -0,0 +1,708 @@
+    }
+
+private:
+    STFUNC(StateWrite) {
+        switch (ev->GetTypeRewrite()) {
+            hFunc(TEvents::TEvCompleted, Handle);
+            hFunc(TEvWorker::TEvHandshake, HoldHandle);
+            hFunc(TEvWorker::TEvData, HoldHandle);
+
+            sFunc(TEvents::TEvPoison, PassAway);
+        }
+    }
+
+    void Handle(TEvents::TEvCompleted::TPtr& ev) {
+        LOG_D("Handle TEvents::TEvCompleted"
+            << ": worker# " << Worker 
+            << " status# " << ev->Get()->Status);
+
+        auto error = TableState->Handle(ev);
+        if (error) {
+            return LogCritAndLeave(error);
+        }
+
+        Send(Worker, new TEvWorker::TEvPoll());
+        return StartWork();
+    }
+
+private:
+
+    TStringBuf GetLogPrefix() const {
+        if (!LogPrefix) {
+            LogPrefix = TStringBuilder()
+                << "[TransferWriter]"
+                << SelfId() << " ";
+        }
+
+        return LogPrefix.GetRef();
+    }
+
+    template <typename TResult>
+    bool CheckResult(const TResult& result, const TStringBuf marker) {
+        if (result.IsSuccess()) {
+            return true;
+        }
+
+        LOG_E("Error at '" << marker << "'"
+            << ", error# " << result);
+        RetryOrLeave(result.GetError());
+
+        return false;
+    }
+
+    void Retry() {
+        Delay = Min(Delay * ++Attempt, MaxDelay);
+        const TDuration random = TDuration::FromValue(TAppData::RandomProvider->GenRand64() % Delay.MicroSeconds());
+        this->Schedule(Delay + random, new TEvents::TEvWakeup());
+    }
+
+    template <typename... Args>
+    void Leave(Args&&... args) {
+        LOG_I("Leave");
+
+        Send(Worker, new TEvWorker::TEvGone(std::forward<Args>(args)...));
+        PassAway();
+    }
+
+    void PassAway() override {
+        TActor::PassAway();
+    }
+
+public:
+    static constexpr NKikimrServices::TActivity::EType ActorActivityType() {
+        return NKikimrServices::TActivity::REPLICATION_S3_WRITER;
+    }
+
+    explicit TTransferWriter(
+        const TString& transformLambda,
+        const TPathId& tablePathId,
+        const TActorId& compileServiceId)
+        : TransformLambda(transformLambda)
+        , TablePathId(tablePathId)
+        , CompileServiceId(compileServiceId)
+        
+    {}
+
+private:
+    const TString TransformLambda;
+    const TPathId TablePathId;
+    const TActorId CompileServiceId;
+    TActorId Worker;
+
+    ITableKindState::TPtr TableState;
+
+    size_t InFlightCompilationId = 0;
+    TProgramHolder::TPtr ProgramHolder;
+
+    mutable TMaybe<TString> LogPrefix;
+
+    TEvWorker::TEvHandshake::TPtr HandshakeEv;
+    TEvWorker::TEvData::TPtr DataEv;
+
+    ui32 Attempt = 0;
+    TDuration Delay = TDuration::Minutes(1);
+    static constexpr TDuration MaxDelay = TDuration::Minutes(10);
+
+}; // TTransferWriter
+
+IActor* CreateTransferWriter(const TString& transformLambda, const TPathId& tablePathId,
+    const TActorId& compileServiceId) {
```

Comment:
```suggestion
IActor* CreateTransferWriter(const TString& transformLambda, const TPathId& tablePathId,
        const TActorId& compileServiceId)
{
```

Или каждый аргумент на новой строке.